src/track-scroll-depth.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { EventTimer } from './EventTimer';
+import { initTrackScrollDepth } from './track-scroll-depth';
+
+describe('initTrackScrollDepth', () => {
+	const observe = jest.fn();
+	class IntersectionObserver {
+		constructor() {
+			return Object.freeze({
+				observe: observe,
```

Comment:
Might be able to inline this:

```suggestion
				observe,
```